internal/transport/proxy_test.go

```diff
@@ -157,6 +176,18 @@ func testHTTPConnect(t *testing.T, proxyURLModify func(*url.URL) *url.URL, proxy
 	if string(recvBuf) != string(msg) {
 		t.Fatalf("received msg: %v, want %v", recvBuf, msg)
 	}
+
+	c.SetReadDeadline(time.Now().Add(20 * time.Millisecond))
+
+	gotServerMessage := make([]byte, len(serverMessage))
+	// This call will return a deadline exceeded error if the server has nothing
+	// to send. This is expected.
+	if _, err := c.Read(gotServerMessage); err != nil {
+		t.Logf("Got error while reading message from server: %v", err)
+	}
```

Comment:
Similar; I don't really like adding a magic deadline to things that we know will timeout sometimes.

Probably we should only do this read if `len(serverMessage) != 0`, and then we can apply a longer deadline while not impacting code that doesn't use it?

Reply:
Changed, not the proxy is configured to wait for server hello.